### ISSUE TITLE
Insert panic due to mismatch of struct schema

### ISSUE THREAD

**Issue by neverchanje** [risingwavelabs/risingwave#4280]
**Describe the bug**

When the struct schema of the insert into statement mismatches with the column schema, it panics.

**To Reproduce**
Steps to reproduce the behavior.

```
CREATE TABLE twitter (
    author STRUCT < created_at TIMESTAMP,
    id VARCHAR,
    name VARCHAR,
    username VARCHAR,
    followers INT >
);

INSERT INTO
    twitter (data, author)
values
    (
        (
            '2019-10-08 10:13:36.07' :: TIMESTAMP,
            '9102642719',
            'Maia Bradtke early',
            'Effertz8562'
        )
    );
```

Panic:

```
thread 'tokio-runtime-worker' panicked at 'itertools: .zip_eq() reached end of one iterator before the other', /Users/wutao/.cargo/registry/src/github.com-1ecc6299db9ec823/itertools-0.10.3/src/zip_eq_impl.rs:48:13
stack backtrace:
   0: std::panicking::begin_panic
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/std/src/panicking.rs:616:12
   1: <itertools::zip_eq_impl::ZipEq<I,J> as core::iter::traits::iterator::Iterator>::next
             at /Users/wutao/.cargo/registry/src/github.com-1ecc6299db9ec823/itertools-0.10.3/src/zip_eq_impl.rs:48:13
   2: core::iter::traits::iterator::Iterator::try_fold
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/iter/traits/iterator.rs:2237:29
   3: <core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::try_fold
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/iter/adapters/map.rs:117:9
   4: <core::iter::adapters::GenericShunt<I,R> as core::iter::traits::iterator::Iterator>::try_fold
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/iter/adapters/mod.rs:191:9
   5: core::iter::traits::iterator::Iterator::try_for_each
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/iter/adapters/mod.rs:174:9
   6: <core::iter::adapters::GenericShunt<I,R> as core::iter::traits::iterator::Iterator>::next
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/iter/adapters/mod.rs:174:9
   7: alloc::vec::Vec<T,A>::extend_desugared
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/alloc/src/vec/mod.rs:2748:35
   8: <alloc::vec::Vec<T,A> as alloc::vec::spec_extend::SpecExtend<T,I>>::spec_extend
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/alloc/src/vec/spec_extend.rs:18:9
   9: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter_nested::SpecFromIterNested<T,I>>::from_iter
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/alloc/src/vec/spec_from_iter_nested.rs:43:9
  10: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter::SpecFromIter<T,I>>::from_iter
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/alloc/src/vec/spec_from_iter.rs:33:9
  11: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/alloc/src/vec/mod.rs:2648:9
  12: core::iter::traits::iterator::Iterator::collect
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/result.rs:2092:49
  13: <core::result::Result<V,E> as core::iter::traits::collect::FromIterator<core::result::Result<A,E>>>::from_iter::{{closure}}
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/result.rs:2092:49
  14: core::iter::adapters::try_process
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/iter/adapters/mod.rs:160:17
  15: <core::result::Result<V,E> as core::iter::traits::collect::FromIterator<core::result::Result<A,E>>>::from_iter
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/result.rs:2092:9
  16: core::iter::traits::iterator::Iterator::collect
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/iter/traits/iterator.rs:1836:9
  17: risingwave_frontend::binder::insert::<impl risingwave_frontend::binder::Binder>::change_null_struct_type
             at ./src/frontend/src/binder/insert.rs:151:34
  18: risingwave_frontend::binder::insert::<impl risingwave_frontend::binder::Binder>::cast_on_insert::{{closure}}
             at ./src/frontend/src/binder/insert.rs:134:29
  19: core::iter::adapters::map::map_try_fold::{{closure}}
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/iter/adapters/map.rs:91:28
  20: core::iter::traits::iterator::Iterator::try_fold
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/iter/traits/iterator.rs:2238:21
  21: <core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::try_fold
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/iter/adapters/map.rs:117:9
  22: <core::iter::adapters::GenericShunt<I,R> as core::iter::traits::iterator::Iterator>::try_fold
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/iter/adapters/mod.rs:191:9
  23: core::iter::traits::iterator::Iterator::try_for_each
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/iter/adapters/mod.rs:174:9
  24: <core::iter::adapters::GenericShunt<I,R> as core::iter::traits::iterator::Iterator>::next
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/iter/adapters/mod.rs:174:9
  25: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter_nested::SpecFromIterNested<T,I>>::from_iter
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/alloc/src/vec/spec_from_iter_nested.rs:26:32
  26: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter::SpecFromIter<T,I>>::from_iter
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/alloc/src/vec/spec_from_iter.rs:33:9
  27: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/alloc/src/vec/mod.rs:2648:9
  28: core::iter::traits::iterator::Iterator::collect
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/result.rs:2092:49
  29: <core::result::Result<V,E> as core::iter::traits::collect::FromIterator<core::result::Result<A,E>>>::from_iter::{{closure}}
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/result.rs:2092:49
  30: core::iter::adapters::try_process
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/iter/adapters/mod.rs:160:17
  31: <core::result::Result<V,E> as core::iter::traits::collect::FromIterator<core::result::Result<A,E>>>::from_iter
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/result.rs:2092:9
  32: core::iter::traits::iterator::Iterator::collect
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/iter/traits/iterator.rs:1836:9
  33: itertools::Itertools::try_collect
             at /Users/wutao/.cargo/registry/src/github.com-1ecc6299db9ec823/itertools-0.10.3/src/lib.rs:2014:9
  34: risingwave_frontend::binder::insert::<impl risingwave_frontend::binder::Binder>::cast_on_insert
             at ./src/frontend/src/binder/insert.rs:130:24
  35: risingwave_frontend::binder::values::<impl risingwave_frontend::binder::Binder>::bind_values::{{closure}}
             at ./src/frontend/src/binder/values.rs:67:32
  36: core::iter::adapters::map::map_try_fold::{{closure}}
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/iter/adapters/map.rs:91:28
  37: core::iter::traits::iterator::Iterator::try_fold
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/iter/traits/iterator.rs:2238:21
  38: <core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::try_fold
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/iter/adapters/map.rs:117:9
  39: <core::iter::adapters::GenericShunt<I,R> as core::iter::traits::iterator::Iterator>::try_fold
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/iter/adapters/mod.rs:191:9
  40: <I as alloc::vec::in_place_collect::SpecInPlaceCollect<T,I>>::collect_in_place
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/alloc/src/vec/in_place_collect.rs:251:13
  41: alloc::vec::in_place_collect::<impl alloc::vec::spec_from_iter::SpecFromIter<T,I> for alloc::vec::Vec<T>>::from_iter
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/alloc/src/vec/in_place_collect.rs:178:19
  42: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/alloc/src/vec/mod.rs:2648:9
  43: core::iter::traits::iterator::Iterator::collect
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/result.rs:2092:49
  44: <core::result::Result<V,E> as core::iter::traits::collect::FromIterator<core::result::Result<A,E>>>::from_iter::{{closure}}
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/result.rs:2092:49
  45: core::iter::adapters::try_process
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/iter/adapters/mod.rs:160:17
  46: <core::result::Result<V,E> as core::iter::traits::collect::FromIterator<core::result::Result<A,E>>>::from_iter
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/result.rs:2092:9
  47: core::iter::traits::iterator::Iterator::collect
             at /rustc/263edd43c5255084292329423c61a9d69715ebfa/library/core/src/iter/traits/iterator.rs:1836:9
  48: itertools::Itertools::try_collect
             at /Users/wutao/.cargo/registry/src/github.com-1ecc6299db9ec823/itertools-0.10.3/src/lib.rs:2014:9
  49: risingwave_frontend::binder::values::<impl risingwave_frontend::binder::Binder>::bind_values
             at ./src/frontend/src/binder/values.rs:65:25
  50: risingwave_frontend::binder::insert::<impl risingwave_frontend::binder::Binder>::bind_insert
             at ./src/frontend/src/binder/insert.rs:82:30
  51: risingwave_frontend::binder::statement::<impl risingwave_frontend::binder::Binder>::bind_statement
             at ./src/frontend/src/binder/statement.rs:38:17
  52: risingwave_frontend::binder::Binder::bind
             at ./src/frontend/src/binder/mod.rs:90:9
  53: risingwave_frontend::handler::dml::handle_dml::{{closure}}
             at ./src/frontend/src/handler/dml.rs:36:9
```

**Expected behavior**

**Additional context**

**Comment by xiangjinwu** [comment]
related: #3692

**Comment by xiangjinwu** [comment]
Seems fixed by #4669
